### DISCUSSION TITLE
How to get lsp.dkjson (now that we're using cmake)

### DISCUSSION THREAD

**Discussion by mhwombat** [orbitalquark/textadept#378] [Q&A] [answered]
I'm trying out the latest version of textadept-lsp from the repo, and I'm getting the `module 'lsp.dkjson' not found`. Back when the module was built with make, we had to run `make deps` to get this file. **Is there a cmake command I need to run?** I tried a few guesses, but I'm not familiar with cmake.

**Comment by orbitalquark** [comment, marked as answer]
Running the initial CMake command pulls down the dependency to the local 
build directory, and the CMake install command puts it in the module 
directory. Something like this should work:

     cmake -S . -B build -D CMAKE_INSTALL_PREFIX=.
     cmake --install build

The GitHub Action that produces release archives does something like 
this, and can be helpful in figuring out how to build local copies of 
modules: 
https://github.com/orbitalquark/textadept-lsp/blob/0c11ffd0fc3fb14f94f4ffa18f4a67f9fdf9bfd9/.github/workflows/release.yml#L16-L18

**Comment by mhwombat** [comment]
Thank you! And you've inspired me to experiment with GitHub Actions.
